### PR TITLE
Rf07 tc015 tc ivan misericordia 06

### PR DESCRIPTION
**Pull request by ivanMise**
RF07-TC-015 UI >Registro de Equipamiento > Verificar que la información de un Equipamiento especifico sea mostrado correctamente después de hacer click en el boton Ver (Nombre, Tipo, Imagen)

image

image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 1]
```diff
@@ -13,11 +13,22 @@
     @FindBy(className = "mat-table")
     private WebElement tableEquipments;
 
+    @FindBy(xpath = "/html/body/ssi-root/ssi-equipment/div/div[3]/div/div/mat-table/mat-row[1]/mat-cell[2]/a[1]")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: reduce your xpath filter

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 2]
```diff
@@ -13,11 +13,22 @@
...
+    private WebElement viewEquipmentBtn;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: change to buttonViewEquipment

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java [hunk 3]
```diff
@@ -13,11 +13,22 @@
...
+
     public ListEquipment() {
         CommonEvents.isVisible(btnCrearEquipment);
     }
 
     public boolean isEquipmentListVisible() {
         return CommonEvents.isVisible(tableEquipments);
     }
+
+    public void clickEquipmentViewBtn() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: change to clickButtonEquipmentView

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    private WebElement contentView;
+
+    @FindBy(className = "mat-card-image")
+    private WebElement imageView;
+
+    @FindBy(className = "mat-card-title-group")
+    private WebElement titleView;
+
+    @FindBy(className = "mat-button")
+    private WebElement backBtn;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: change to buttonBack

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    public ViewEquipment() {
+        CommonEvents.isVisible(backBtn);
+    }
+
+    public boolean checkEquipmentView() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: chage to isVisibleEquipmentView

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        return checkContentView() && checkImageView() && checkTitleView();
+    }
+
+    private boolean checkContentView() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: isVisibleContentView

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        return CommonEvents.isVisible(contentView);
+    }
+
+    private boolean checkImageView() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: isVisibleImageView

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        return CommonEvents.isVisible(imageView);
+    }
+
+    private boolean checkTitleView() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: isVisibleTitleView

File: src/test/resources/features/equipments/checkdetailEquipment.feature [hunk 9]
```diff
@@ -0,0 +1,14 @@
+
+Feature: Check Detail Equipment
```
Comment (marbusjim) [inline src/test/resources/features/equipments/checkdetailEquipment.feature]: use spanish

File: src/test/resources/features/equipments/checkdetailEquipment.feature [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+    Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+      Given 'SSI-D' pagina de inicio de sesion es cargada
+      And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+    Scenario: Verificar que el 'Detalle de un Equipamiento' especifico sea mostrado correctamente
+      Given 'Menu principal' estee cargado
+      And seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'
+      And Seleccionar submenu 'Equipamiento' en menu 'Equipamiento'
+      And click en boton Ver de 'Lista de Equipamientos'
```
Comment (marbusjim) [inline src/test/resources/features/equipments/checkdetailEquipment.feature]: click en boton 'Ver' de 'Lista de Equipamientos'

**Review by marbusjim — COMMENTED** [review]
review the comments

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java (hunk 1 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: change xpath by other more short and specific

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java (hunk 2 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: rename webElement Ej.: buttonViewEquipment

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java (hunk 3 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: rename this method Ej.: clickButtonViewEquipmant()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java (hunk 4 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: rename webelement Ej: buttonBack

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java (hunk 6 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: rename this method Ej: isContentViewVisible()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java (hunk 7 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: rename this method Ej: isImageViewVisible()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java (hunk 8 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ViewEquipment.java]: rename this method Ej: isTitleViewVisible()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java (hunk 3 above)
Comment (jdpapiero) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListEquipment.java]: this method should return some pageObject

File: src/test/resources/features/equipments/checkdetailEquipment.feature [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+      When 'Detalle de Equipamiento' correctamente cargado
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkdetailEquipment.feature]: maybe if and previous step you return a page object, this step is not necessary

File: src/test/resources/features/equipments/checkdetailEquipment.feature [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+Feature: Check Detail Equipment
+
+    Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+      Given 'SSI-D' pagina de inicio de sesion es cargada
+      And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+    Scenario: Verificar que el 'Detalle de un Equipamiento' especifico sea mostrado correctamente
+      Given 'Menu principal' estee cargado
+      And seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'
+      And Seleccionar submenu 'Equipamiento' en menu 'Equipamiento'
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkdetailEquipment.feature]: please the first step in lowercase

File: src/test/resources/features/equipments/checkdetailEquipment.feature [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+      And click en boton Ver de 'Lista de Equipamientos'
+      When 'Detalle de Equipamiento' correctamente cargado
+      Then Verificar que el 'Detalle de Equipamiento' especifico sea mostrado correctamente
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkdetailEquipment.feature]: please the first step in lowercase

**Review by jdpapiero — CHANGES_REQUESTED** [review]
review observations

**Review by danielrpg — APPROVED** [review]
Code review aproved

**Review by jdpapiero — APPROVED** [review]
Approved

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 14]
```diff
@@ -261,6 +263,28 @@ public void validarComponenteImagenFormularioRegistroEquipamiento() throws Throw
         Assert.assertTrue(formEquipment.verifyInputImageFormIsVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipments title"));
     }
 
+<<<<<<< HEAD
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: please verify this

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 15]
```diff
@@ -261,6 +263,28 @@ public void validarComponenteImagenFormularioRegistroEquipamiento() throws Throw
...
+    public void verificarTipoEnDetalleEquipamiento() throws Throwable {
+        Assert.assertTrue(equipmentDetail.verifyEquipmentTypeIsVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "EquipmentDetail Title"));
+    }
+
+    @And("^verificar contenido de 'Imagen' en el  'Detalle de Equipamiento' este visible$")
+    public void verificarImagenEnDetalleEquipamiento() throws Throwable {
+        Assert.assertTrue(equipmentDetail.verifyEquipmentImageIsVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "EquipmentDetail Title"));
+    }
+    //END Equipments
+=======
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: please verify this

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 16]
```diff
@@ -281,6 +305,7 @@ public void verificarMensajeDescripcionRequeridoEnRegistroequipamiento() throws
         Assert.assertTrue(formEquipment.verifyHintMessageInputDescriptionIsVisible(), String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Equipment title"));
     }
     //END Equipment
+>>>>>>> 8214145e67ca367ed456e354cfb490e36e62e45e
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: please verify this

**Review by marbusjim — APPROVED** [review]
approve

**Pull request merged**